### PR TITLE
Modify ReportService's get_config_exploits() to use plugin repository

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2857 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > UTs pass
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 1]
```diff
@@ -437,13 +439,23 @@ def get_cross_segment_issues(cls):
...
+            AgentPluginType.EXPLOITER, {}
+        )
+        for exploiter, manifest in plugin_exploiter_manifests.items():
+            if exploiter in exploitation_configuration.exploiters:
+                configured_exploiter_names.append(manifest.title)
+
+        for exploiter, manifest in HARD_CODED_EXPLOITER_MANIFESTS.items():
+            if exploiter in exploitation_configuration.exploiters:
+                configured_exploiter_names.append(manifest.title)
+
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/services/reporting/report.py:457]:
```suggestion
        plugin_exploiter_manifests = cls._agent_plugin_repository.get_all_plugin_manifests().get(
            AgentPluginType.EXPLOITER, {}
        )
        plugin_exploiter_manifests.update(HARD_CODED_EXPLOITER_MANIFESTS)
        for exploiter, manifest in plugin_exploiter_manifests.items():
            if exploiter in exploitation_configuration.exploiters:
                configured_exploiter_names.append(manifest.title)

```
Less lines 😄

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 2]
```diff
@@ -437,13 +439,20 @@ def get_cross_segment_issues(cls):
 
     @classmethod
     def get_config_exploits(cls):
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/reporting/report.py:441]: add typehint

**Review by VakarisZ — APPROVED** [review]

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 3]
```diff
@@ -436,14 +438,24 @@ def get_cross_segment_issues(cls):
...
+        exploiter_manifests = cls._agent_plugin_repository.get_all_plugin_manifests().get(
+            AgentPluginType.EXPLOITER, {}
+        )
+        exploiter_manifests.update(HARD_CODED_EXPLOITER_MANIFESTS)
+        for exploiter, manifest in exploiter_manifests.items():
+            if exploiter in exploitation_configuration.exploiters:
+                if manifest.title:
+                    configured_exploiter_names.append(manifest.title)
+                else:
+                    configured_exploiter_names.append(manifest.name)
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/reporting/report.py:456]: See https://github.com/guardicore/monkey/pull/2876#discussion_r1085309909

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 4]
```diff
@@ -436,14 +438,27 @@ def get_cross_segment_issues(cls):
...
-            for exploiter in exploitation_configuration.exploiters
-        ]
+        exploiter_manifests = cls._agent_plugin_repository.get_all_plugin_manifests().get(
+            AgentPluginType.EXPLOITER, {}
+        )
+        if not exploiter_manifests:
+            logger.debug("No plugin exploiter manifests were found")
+
+        exploiter_manifests.update(HARD_CODED_EXPLOITER_MANIFESTS)
+        for exploiter, manifest in exploiter_manifests.items():
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/reporting/report.py:454]:
```suggestion
        for exploiter_name, manifest in exploiter_manifests.items():
```

File: monkey/monkey_island/cc/services/reporting/report.py [hunk 5]
```diff
@@ -436,14 +438,27 @@ def get_cross_segment_issues(cls):
...
+            if exploiter in exploitation_configuration.exploiters:
+                if manifest.title:
+                    configured_exploiter_names.append(manifest.title)
+                else:
+                    configured_exploiter_names.append(manifest.name)
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/services/reporting/report.py:459]: Style suggestion:
```suggestion
            if exploiter not in exploitation_configuration.exploiters:
                continue
                
            if manifest.title:
                configured_exploiter_names.append(manifest.title)
            else:
                configured_exploiter_names.append(manifest.name)
```

**Review by mssalvatore — APPROVED** [review]

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**